renku/core/dataset/providers/s3.py

```diff
@@ -70,25 +69,55 @@ def supports_add() -> bool:
         """Whether this provider supports adding data to datasets."""
         return True
 
+    @staticmethod
+    def get_add_parameters() -> List["ProviderParameter"]:
+        """Returns parameters that can be set for add."""
+        from renku.core.dataset.providers.models import ProviderParameter
+
+        return [
+            ProviderParameter(
+                "sources",
+                flags=["s", "source"],
+                default=None,
+                help="Path(s) within the s3 bucket that should be added",
+                multiple=True,
+                type=str,
+            ),
```

Comment:
I'm not sure whats the benefit of this option; we can simply pass the value that we pass here as a URI and things should work as normal. As I understood, this is supposed to act as a prefix to URI so that if we run `renku dataset add s3 -s s3://giab/ Aspera_download_from_ftp.README` we would get `s3://giab/Aspera_download_from_ftp.README`. However, this won't work as intended because in this case URI is `Aspera_download_from_ftp.README` which won't be processed by the S3 provider. Running the above command result in:
```
Error: Invalid parameter value - The scheme of some urls ('Aspera_download_from_ftp.README',) does not match the defined storage url s3://giab/.
```

Reply:
@mohammad-sdsc I have removed the --source option for s3 datasets as you suggested. It is def. better to simply not allow this.